service/resource/vaultcrt/create.go

```diff
@@ -0,0 +1,123 @@
+package vaultcrt
+
+import (
+	"context"
+
+	"github.com/giantswarm/certificatetpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/vaultcrt"
+	"github.com/giantswarm/vaultrole"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentSecret, err := toSecret(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredSecret, err := toSecret(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the secret has to be created")
+
+	var secretToCreate *apiv1.Secret
+	if currentSecret == nil {
+		secretToCreate = desiredSecret
+
+		err := r.ensureVaultRole(customObject)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		ca, crt, key, err := r.issueCertificate(customObject)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		secretToCreate.StringData[certificatetpr.CA.String()] = ca
+		secretToCreate.StringData[certificatetpr.Crt.String()] = crt
+		secretToCreate.StringData[certificatetpr.Key.String()] = key
+	}
```

Comment:
Note that we create a role per key-pair now with potential support for organizations. We just do not use this support now because the organizations information is not yet given with the custom object.

Reply:
I don't follow here - what does this have to do with roles?

Reply:
This is just how Vault works. There has to be a Vault role for each certificate we want to issue via the Vault PKI backend.